ndc-models/src/relational_query/expression.rs

```diff
@@ -0,0 +1,911 @@
+    /// * During filtering: `relational_query.filter.scalar.cos`
+    /// * During sorting:`relational_query.sort.expression.scalar.cos`
+    /// * During joining: `relational_query.join.expression.scalar.cos`
+    /// * During aggregation: `relational_query.aggregate.expression.scalar.cos`
+    /// * During windowing: `relational_query.window.expression.scalar.cos`
+    Cos {
+        expr: Box<RelationalExpression>,
+    },
+    /// Only used when in specific contexts where the appropriate capability is supported:
+    /// * During projection: `relational_query.project.expression.scalar.current_date`
+    /// * During filtering: `relational_query.filter.scalar.current_date`
+    /// * During sorting:`relational_query.sort.expression.scalar.current_date`
+    /// * During joining: `relational_query.join.expression.scalar.current_date`
+    /// * During aggregation: `relational_query.aggregate.expression.scalar.current_date`
+    /// * During windowing: `relational_query.window.expression.scalar.current_date`
+    CurrentDate,
+    /// Only used when in specific contexts where the appropriate capability is supported:
+    /// * During projection: `relational_query.project.expression.scalar.current_time`
+    /// * During filtering: `relational_query.filter.scalar.current_time`
+    /// * During sorting:`relational_query.sort.expression.scalar.current_time`
+    /// * During joining: `relational_query.join.expression.scalar.current_time`
+    /// * During aggregation: `relational_query.aggregate.expression.scalar.current_time`
+    /// * During windowing: `relational_query.window.expression.scalar.current_time`
+    CurrentTime,
+    /// Only used when in specific contexts where the appropriate capability is supported:
+    /// * During projection: `relational_query.project.expression.scalar.current_timestamp`
+    /// * During filtering: `relational_query.filter.scalar.current_timestamp`
+    /// * During sorting:`relational_query.sort.expression.scalar.current_timestamp`
+    /// * During joining: `relational_query.join.expression.scalar.current_timestamp`
+    /// * During aggregation: `relational_query.aggregate.expression.scalar.current_timestamp`
+    /// * During windowing: `relational_query.window.expression.scalar.current_timestamp`
+    CurrentTimestamp,
+    /// Only used when in specific contexts where the appropriate capability is supported:
+    /// * During projection: `relational_query.project.expression.scalar.date_part`
+    /// * During filtering: `relational_query.filter.scalar.date_part`
+    /// * During sorting:`relational_query.sort.expression.scalar.date_part`
+    /// * During joining: `relational_query.join.expression.scalar.date_part`
+    /// * During aggregation: `relational_query.aggregate.expression.scalar.date_part`
+    /// * During windowing: `relational_query.window.expression.scalar.date_part`
+    DatePart {
+        expr: Box<RelationalExpression>,
+        part: Box<RelationalExpression>,
```

Comment:
This can only be a static string, and one of the same strings as used in `extract` in the group by part of the spec.

Reply:
I've added an enum and related capabilities for this.